lib/__tests__/process.test.ts

```diff
@@ -0,0 +1,146 @@
+import readline from 'readline';
+import { logger, setLogLevel, LOG_LEVEL } from '@hubspot/local-dev-lib/logger';
+import { handleExit, handleKeypress } from '../process';
+
+jest.mock('readline');
+jest.mock('@hubspot/local-dev-lib/logger');
+
+const mockedReadline = readline as jest.Mocked<typeof readline>;
+const mockedLogger = logger as jest.Mocked<typeof logger>;
+const mockedSetLogLevel = setLogLevel as jest.Mock;
+const processRemoveListenerSpy = jest.spyOn(process, 'removeAllListeners');
+const processOnSpy = jest.spyOn(process, 'on');
+
+describe('lib/process', () => {
+  describe('handleExit()', () => {
+    const mockCallback = jest.fn();
+    const terminationSignals = [
+      'beforeExit',
+      'SIGINT',
+      'SIGUSR1',
+      'SIGUSR2',
+      'uncaughtException',
+      'SIGTERM',
+      'SIGHUP',
+    ];
+
+    it('should set up listeners for all termination signals', () => {
+      handleExit(mockCallback);
+
+      terminationSignals.forEach(signal => {
+        expect(processRemoveListenerSpy).toHaveBeenCalledWith(signal);
+        expect(processOnSpy).toHaveBeenCalledWith(signal, expect.any(Function));
+      });
+    });
+
+    it('should handle SIGHUP signal correctly', async () => {
+      handleExit(mockCallback);
+
+      // Get the callback function passed to process.on for SIGHUP
+      const sighupCallback = processOnSpy.mock.calls.find(
+        call => call[0] === 'SIGHUP'
+      )?.[1];
+
+      if (!sighupCallback) {
+        throw new Error('SIGHUP callback not found');
+      }
```

Comment:
Nit, instead of throwing we should just `expect(sighupCallback).toBeDefined()`

Reply:
done in [50ae059](https://github.com/HubSpot/hubspot-cli/pull/1377/commits/50ae059b50dd10cdfbd03e26b7f9120273eb7ebd) 👍